src/log.action.ts

```diff
@@ -1,14 +1,33 @@
-import { ActionResponse, After, Before, flat } from 'adminjs';
-import { merge } from 'lodash';
+import {
+  ActionContext,
+  ActionRequest,
+  ActionResponse,
+  After,
+  Before,
+  flat,
+} from 'adminjs';
 
 import { difference } from './utils/difference';
 import { getLogPropertyName } from './utils/get-log-property-name';
 import { LoggerFeatureOptions } from './logger.feature';
 
-export const rememberInitialRecord: Before = async (request, context) => {
-  const id = context.record?.id();
+export const rememberInitialRecord: Before = async (
+  request: ActionRequest,
+  context: ActionContext
+) => {
+  if (request.params.action === 'bulkDelete') {
+    context.initialRecords = request.query?.recordIds
+      ? await Promise.all(
+          request.query.recordIds
+            .split(',')
+            .map(id => context.resource.findOne(id))
```

Comment:
maybe we can use `resource.findMany`? it should have better database performance